### PR TITLE
PP-5115: Add ledger Jenkinsfile build configuration

### PR DESCRIPTION
**Pull request by rjbaker**
### WHAT:
Adds a Jenkinsfile to build, test and (later) deploy the ledger service.

### How to test & deploy:
The Jenkins CI configuration must be updated in alphagov/pay-chef to scan and build this repository. Once this has been deployed to CI, this application may be built.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+          }
+        }
+      }
+      post {
+        failure {
+          postMetric("ledger.docker-build.failure", 1)
+        }
+      }
+    }
+    stage('Tests') {
```
Comment (mrlumbu) [inline Jenkinsfile]: we don't need any e2e tests to start with as none of our current e2e tests will touch this service. We can change this as and when we write those tests

Comment (mrlumbu) [reply Jenkinsfile]: i.e. you can remove this whole stage for now

Comment (rjbaker) [reply Jenkinsfile]: Thanks. No simple way to skip. Have removed for now. 👍

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        }
+      }
+    }
+    stage('Deploy') {
+      when {
+        branch 'master'
+      }
+      steps {
+        checkPactCompatibility("ledger", gitCommit(), "test")
+        deployEcs("ledger")
```
Comment (whpearson) [inline Jenkinsfile]: This will attempt to deploy to ecs, so will fail for now.
